engine/src/main/java/io/zeebe/engine/processor/workflow/job/JobBatchActivateProcessor.java

```diff
@@ -144,6 +153,12 @@ private void collectJobsToActivate(
             arrayValueJob.wrap(buffer);
           } else {
             value.setTruncated(true);
+
+            if (value.getJobs().isEmpty()) {
+              raiseIncidentJobTooLargeForMessageSize(key, jobRecord, streamWriter);
+              jobState.disable(key, jobRecord);
+            }
```

Comment:
I agree with you that it would be easier to follow. However, then I would not have access to the `jobRecord`.

Would you rather leave it as it is, or refactor it for the collect method to return something that then can be analyzed?
Or am I missing something?

Reply:
Good point. Then I don't have a suggestion.